k8s_test_harness/util/docker_util.py

```diff
@@ -180,3 +181,28 @@ def get_image_version(image):
         text=True,
     )
     return process.stdout.strip()
+
+
+def run_entrypoint_and_assert(
+    image, entrypoint, expect_stdout_contains=None, expect_stderr_contains=None
+):
+    """Runs the given entrypoint in the given image and asserts that it succeeds.
+
+    If expect_stdout_contains is provided, also asserts that the given string
+    is contained in the stdout. Similarly for expect_stderr_contains.
+    """
+    entry = shlex.split(entrypoint) if isinstance(entrypoint, str) else entrypoint
+    process = run_in_docker(image, entry, check_exit_code=False)
+    assert (
+        process.returncode == 0
+    ), f"Failed to run {entry} in image {image}, stderr: {process.stderr}"
```

Comment:
Makes reading a bit easier

```suggestion
    assert (
        process.returncode == 0
    ), f"Failed to run `{shlex.join(entry)}` in image {image}, {process.stdout=} {process.stderr=}"
```